TypeScript can't find types when importing `cbor-x/index-no-eval`

Hello @kriszyp and thank you for your excellent work with cbor-x! I come with a bit of an odd problem: TypeScript doesn't understand where to find types when importing things from `cbor-x/index-no-eval`:

**cborx-test.ts**
```ts
import { Encoder } from 'cbor-x/index-no-eval';

const encoder = new Encoder();
```

**tsconfig.json**
```json
{
  "compilerOptions": {
    "target": "ES2018",
    "lib": ["ES2018", "DOM"],
    "module": "commonjs",
    "esModuleInterop": true,
    "declaration": true,
    "sourceMap": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "outDir": "dist"
  },
  "exclude": [
    "node_modules",
    "**/dist/*"
  ]
}
```

```sh
# TypeScript 4.6.3
$> npx tsc cborx-test.ts
cborx-test.ts:1:25 - error TS2307: Cannot find module 'cbor-x/index-no-eval'
or its corresponding type declarations.

1 import { Encoder } from 'cbor-x/index-no-eval';
                          ~~~~~~~~~~~~~~~~~~~~~~
```

I found out that if I duplicate **index.d.ts** as **index-no-eval.d.ts** in this project (within my local **node_modules/**) then subsequent runs of `tsc` succeed.

I wanted to PR something but after realizing that **index.d.ts** is manually maintained over here I couldn't think of anything clever to propose as a solution. And I didn't want to propose anything that involved maintaining a separate copy of **index.d.ts** as **index-no-eval.d.ts** as that's proposing too much maintenance and potential for type desync causing more headaches later.

Thanks for considering this.

--- a/package.json
+++ b/package.json
@@ -55,8 +55,14 @@
         "import": "./decode.js"
       }
     },
-    "./decode-no-eval": "./dist/decode-no-eval.cjs",
-    "./index-no-eval": "./dist/index-no-eval.cjs",
+    "./decode-no-eval": {
+      "types": "./decode.d.ts",
+      "default": "./dist/decode-no-eval.cjs"
+    },
+    "./index-no-eval": {
+      "types": "./index.d.ts",
+      "default": "./dist/index-no-eval.cjs"
+    },
     "./package.json": "./package.json"
   },
   "files": [

